src/liquid/components/ld-sidenav/ld-sidenav-accordion/ld-sidenav-accordion.tsx

```diff
@@ -0,0 +1,163 @@
+  handleSidenavBreakpointChange(ev: CustomEvent<boolean>) {
+    if (ev.target !== this.sidenav) return
+    this.sidenavClosable = ev.detail
+  }
+
+  @Listen('ldSidenavSliderChange', { target: 'window', passive: true })
+  handleSidenavSliderChange(ev: CustomEvent<boolean>) {
+    // Collapse accordion on sidenav slide change.
+    if (
+      (ev.target as HTMLLdSidenavSliderElement).closest('ld-sidenav') !==
+      this.sidenav
+    ) {
+      return
+    }
+    if (!this.preserveState) {
+      this.sectionRef.expanded = false
+      this.expanded = false
+    }
+    this.transitionsEnabled = false
+    setTimeout(() => {
+      this.transitionsEnabled = true
+    }, 10)
+  }
+
+  @Listen('ldSidenavCollapsedChange', { target: 'window', passive: true })
+  handleSidenavCollapsedChange(ev: CustomEvent<boolean>) {
+    // Collapse or expand accordion on sidenav collapse or expansion.
+    if (ev.target !== this.sidenav) return
+    this.sidenavCollapsed = ev.detail
+    if (this.sidenav.narrow) {
+      toggleStackToTop(this.el, this.sidenavCollapsed)
+    }
+    if (this.sidenavCollapsed) {
+      if (this.preserveState) {
+        this.expandOnSidenavExpansion = this.sectionRef.expanded
+      }
+      this.sectionRef.expanded = false
+      this.expanded = false
+    } else {
+      if (this.expandOnSidenavExpansion) {
+        this.sectionRef.expanded = true
+        this.expanded = true
+      }
+    }
+  }
+
+  @Watch('expanded')
+  handleExpandedChange(expanded) {
+    this.sectionRef.expanded = expanded
+  }
+
+  private onTransitionEnd = (ev: TransitionEvent) => {
+    // Emit transition end event to trigger scroll shadow update.
+    if (ev.target === this.panelRef) {
+      this.ldSidenavAccordionTransitionEnd.emit()
+    }
+  }
+
+  componentWillLoad() {
+    this.inAccordion = this.el.parentElement.tagName === 'LD-SIDENAV-ACCORDION'
+    this.sidenav = closest('ld-sidenav', this.el)
+  }
+
+  render() {
+    const cl = getClassNames([
+      'ld-sidenav-accordion',
+      this.inAccordion && 'ld-sidenav-accordion--in-accordion',
+      this.sidenavCollapsed &&
+        !this.sidenavClosable &&
+        'ld-sidenav-accordion--collapsed',
+      this.split && 'ld-sidenav-accordion--split',
+    ])
+
+    return (
+      <Host class={cl}>
+        <ld-accordion class="ld-sidenav-accordion__accordion">
+          <ld-accordion-section
+            ref={(el) => (this.sectionRef = el)}
+            class="ld-sidenav-accordion__accordion-section"
+          >
+            <ld-accordion-toggle
+              labelTag="div"
+              class="ld-sidenav-accordion__accordion-toggle"
+              split={this.split}
+            >
+              <slot name="toggle"></slot>
+            </ld-accordion-toggle>
+            <ld-accordion-panel
+              style={
+                this.transitionsEnabled
+                  ? undefined
+                  : { '--ld-accordion-panel-transition-duration': '0.0001s' }
+              }
```

Comment:
Why not using a modifier class for `ld-sidenav-accordion__accordion-panel` instead of inline styles?